src/socketfactory.js

```diff
@@ -0,0 +1,238 @@
+var EventEmitter = require('events').EventEmitter;
```

Comment:
It might be good to add a short comment at the top of the file explaining the public interface here.  It took a few mins of reading to figure out that callers don't actually use the SocketPool, rather they just call SocketFactory.getSocket, which returns a Socket object, whose only methods are write and close, and it emits 'onData' and 'onDisconnect' events